diesel_tests/tests/debug/mod.rs

```diff
@@ -0,0 +1,35 @@
+use diesel::*;
+use diesel::query_builder::debug::DebugQueryBuilder;
+use diesel::query_builder::update;
+
+#[test]
+fn test_debug_count_output() {
+    use schema::users::dsl::*;
+    let mut query_builder = DebugQueryBuilder::new();
+    let command = users.count();
+    command.to_sql(&mut query_builder).unwrap();
+    assert_eq!(query_builder.sql, "SELECT COUNT(*) FROM users");
+}
+
+#[test]
+fn test_debug_output() {
+    use schema::users::dsl::*;
+    use diesel::query_builder::QueryFragment;
+    let mut query_builder = DebugQueryBuilder::new();
+    let command = update(users.filter(id.eq(1))).set(name.eq("new_name"));
+    command.to_sql(&mut query_builder).unwrap();
+    assert_eq!(
+      query_builder.sql,
+      "UPDATE users SET name = $1 WHERE users.id = $2"
+    );
+}
+
+// run this with cargo test -- --nocapture
```

Comment:
We can't test it. Let's have the definition of `print_sql!` be `println!("{}", &debug_sql!(args))`, and have `debug_sql!` return a `String`. Then we can test `debug_sql` which is 99% of the behavior.